asdf/asdf.py

```diff
@@ -92,6 +92,12 @@ def __init__(self, tree=None, uri=None, extensions=None, version=None,
             When `False`, when reading files, attempt to memmap underlying data
             arrays when possible.
 
+        lazy_load : bool, optional
+            When `True` and the read file is seekable, underlying data arrays
+            will be materialized when they are used for the first time.
+            This implies that the file should stay open during the lifetime
+            of the tree.
```

Comment:
Sorry to nitpick, but I find this a little confusing. Instead I might say the following:

"When `True` and the underlying file handle is seekable, data arrays will only be loaded lazily: i.e. when they are accessed for the first time. In this case the underlying file must stay open during the lifetime of the tree. Setting to `False` causes all data arrays to be loaded up front, which means that they can be accessed even after the underlying file is closed."

I'm happy to make this change myself, BTW, but just want to make sure it makes sense to you.

Reply:
So much better!